src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java

```diff
@@ -175,25 +187,94 @@ public DbfsService getDbfsService() {
    */
   public static class Builder {
 
-    long retryInterval = DEFAULT_HTTP_CLIENT_RETRY_INTERVAL;
-    int maxRetries = DEFAULT_HTTP_CLIENT_MAX_RETRY;
-    String token;
     String host;
+    String token;
     String username;
     String password;
 
+    /**
+     * Databricks rest-api version.
+     */
+    String apiVersion = "2.0";
+
+    /**
+     * Databricks rest http client {@link com.edmunds.rest.databricks.HttpServiceUnavailableRetryStrategy} default.
+     */
+    long retryInterval = DEFAULT_HTTP_CLIENT_RETRY_INTERVAL;
+    int maxRetries = DEFAULT_HTTP_CLIENT_MAX_RETRY;
+
+    /**
+     * Databricks rest http client socket parameters.
+     */
+    int soTimeout = SOCKET_TIMEOUT;
+    int connectionTimeout = CONNECTION_TIMEOUT;
+    int connectionRequestTimeout = CONNECTION_REQUEST_TIMEOUT;
+
+
+    /**
+     * When use databricks service on CDH 5.7.1 , useLegacyAPI425 set true for v4.2.5 compatible API.
```

Comment:
this comment about cdh5.7.1 will confuse open source audience

Reply:
Changed comment to more genral way